File: api/api.web/Controllers/SearchController.fs
namespace Djambi.Api.Web.Controllers

open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open FSharp.Control.Tasks
open Serilog
open Djambi.Api.Logic.Interfaces
open Djambi.Api.Web
open Djambi.Api.Web.Model
open Djambi.Api.Web.Mappings

[<ApiController>]
[<Route("api/search")>]
type SearchController(manager : ISearchManager,
                       logger : ILogger,
                       scp : SessionContextProvider) =
    inherit ControllerBase()
    
    [<HttpPost("games")>]
    [<ProducesResponseType(200, Type = typeof<SearchGameDto[]>)>]
    member __.SearchGames([<FromBody>] query : GamesQueryDto) : Task<IActionResult> =
        let ctx = base.HttpContext
        task {
            let! session = scp.GetSessionFromContext ctx
            let query = query |> toGamesQuery
            let! games = manager.searchGames query session
            let dtos = games |> List.map toSearchGameDto
            return OkObjectResult(dtos) :> IActionResult
        }

File: web2/src/controllers/searchController.ts
import { GamesQueryDto, SearchGameDto, ApiSearchGamesPostRequest } from '../api-client';
import * as Api from '../utilities/api';

export async function searchGames(query: GamesQueryDto): Promise<SearchGameDto[]> {
  const request: ApiSearchGamesPostRequest = { gamesQueryDto: query };
  const games = await Api.search().apiSearchGamesPost(request);
  return games;
}
